nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,415 @@
+            data of both hemispheres/parts.
+            shape: (n_vertices, n_regions)
+
+        mask_img_ : :obj:`~numpy.ndarray` or None
+            The mask image converted to a numpy array by concatenating the \
+            `mask_img` data of both hemispheres/parts.
+            shape: (n_vertices,)
+
+        n_elements_ : :obj:`int`
+            The number of regions in the maps image.
+
+    See Also
+    --------
+        nilearn.maskers.SurfaceMasker
+        nilearn.maskers.SurfaceLabelsMasker
+
+    """
+
+    def __init__(
+        self,
+        maps_img,
+        mask_img=None,
+        allow_overlap=True,
+        smoothing_fwhm=None,
+        standardize=False,
+        standardize_confounds=True,
+        detrend=False,
+        high_variance_confounds=False,
+        low_pass=None,
+        high_pass=None,
+        t_r=None,
+        memory=None,
+        memory_level=1,
+        verbose=0,
+        reports=True,
+        cmap="inferno",
+        clean_args=None,
+    ):
+        self.maps_img = maps_img
+        self.mask_img = mask_img
+        self.allow_overlap = allow_overlap
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.standardize_confounds = standardize_confounds
+        self.high_variance_confounds = high_variance_confounds
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.reports = reports
+        self.cmap = cmap
+        self.clean_args = clean_args
+        self._shelving = False
+        # content to inject in the HTML template
+        self._report_content = {
+            "description": (
+                "This report shows the input surface image "
+                "(if provided via img) overlaid with the regions provided via "
+                "maps_img."
+            ),
+            "n_vertices": {},
+            "number_of_regions": 0,
+            "summary": {},
+        }
+
+    def fit(self, img=None, y=None):
+        """Prepare signal extraction from regions.
+
+        Parameters
+        ----------
+        img : :obj:`~nilearn.surface.SurfaceImage` object or None, default=None
+            This parameter is currently unused.
+
+        y : None
+            This parameter is unused.
+            It is solely included for scikit-learn compatibility.
+
+        Returns
+        -------
+        SurfaceMapsMasker object
+        """
+        del img, y
+
+        logger.log(
+            msg=f"loading regions from {self.maps_img.__repr__()}",
+            verbose=self.verbose,
+        )
+        # check maps_img data is 2D
+        check_surface_data_ndims(self.maps_img, 2, "maps_img")
+
+        self.maps_img_ = np.concatenate(
+            list(self.maps_img.data.parts.values()), axis=0
+        )
```

Comment:
TODO
use the concatenate function from nilearn.maskers._utils

Reply:
But that one concatenates different surface images provided in a list. 

Here I want to concatenate the data from two hemispheres of the _same_ surface image for later computation.

Reply:
my bad